src/idl_gen_rust.cpp

```diff
@@ -1327,12 +1327,21 @@ class RustGenerator : public BaseGenerator {
         code_ += "  #[allow(non_snake_case)]";
         code_ += "  pub fn {{FIELD_NAME}}_as_{{U_ELEMENT_NAME}}(&'a self) -> "
                  "Option<{{U_ELEMENT_TABLE_TYPE}}> {";
-        code_ += "    if self.{{FIELD_NAME}}_type() == {{U_ELEMENT_ENUM_TYPE}} {";
-        code_ += "      self.{{FIELD_NAME}}().map(|u| "
-                 "{{U_ELEMENT_TABLE_TYPE}}::init_from_table(u))";
-        code_ += "    } else {";
-        code_ += "      None";
-        code_ += "    }";
+
+        if (field.required) {
+          code_ += "      None";
+        } else {
+          code_ += "    if self.{{FIELD_NAME}}_type() == {{U_ELEMENT_ENUM_TYPE}} {";
+          code_ += "      self.{{FIELD_NAME}}().map(|u| "
```

Comment:
This is where the conditional on `required` needs to be: if the field is `required`, don't do the `Option` `map`.

Reply:
Now, If the field is required, it will skip the `.map` and return `Some(U_ELEMENT_TABLE_TYPE)`